starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/OperatorAutoConfiguration.java

```diff
@@ -121,6 +125,45 @@ public Operator operator(
     return operator;
   }
 
+  @SuppressWarnings("rawtypes")
+  private void setControllerOverrides(ControllerConfigurationOverrider<?> o,
```

Comment:
Thinking more about it, the starter should either reuse an implementation of `ControllerConfiguration` or provide its own instead of relying on overriding.

Reply:
Why? I think this is much more elegant, basically that is what you want to do, override the defaults by providing properties.

Reply:
Ideally starter is just a wrapper, we should avoid having custom implementations if not really needed.

Reply:
It looks more elegant but also wrong as the resource type shouldn't be changed from what the `Reconciler` implementation defines.

Reply:
We can emit resource type, and add it if somebody asks for it. And see the reasons.